src/logic/referral/referral.ts

```diff
@@ -367,6 +440,12 @@ export async function createReferralComponent(
         throw new ReferralInvalidInputError(`You must have at least ${TIERS_IRL_SWAG} accepted invites to set an email`)
       }
 
+      const denyList = await fetchDenyList()
+
+      if (denyList.has(referrer.toLowerCase())) {
```

Comment:
Can we move this line to be above of the function call:
`const acceptedInvites = await referralDb.countAcceptedInvitesByReferrer(referrer)`

So if it is a blacklisted referrer, we don't even go to the database